src/main/java/org/sagebionetworks/bridge/services/Exporter3Service.java

```diff
@@ -968,66 +971,60 @@ private void exportUpload(String appId, String recordId) {
     // so they need access to the Timeline information.).
     public Exporter3Configuration exportTimelineForStudy(String appId, String studyId) throws BridgeSynapseException,
             SynapseException, IOException{
+        boolean isConfigModified = false;
 
         // Get Timeline to export for the study.
+        Study study = studyService.getStudy(appId, studyId, true);
+        if (study.getScheduleGuid() == null) {
+            throw new EntityNotFoundException(Schedule2.class);
+        }
         Timeline timeline = schedule2Service.getTimelineForSchedule(appId,
-                studyService.getStudy(appId, studyId, true).getScheduleGuid());
+                study.getScheduleGuid());
 
         // Check Synapse
         synapseHelper.checkSynapseWritableOrThrow();
 
-        // Initiate Exporter3 for Study.
-        Exporter3Configuration exporter3Config = initExporter3ForStudy(appId, studyId);
+        // If Exporter3 is not enabled for study, initiate Exporter3 for Study;
+        // If enabled, get exporter3Config for study.
+        Exporter3Configuration exporter3Config;
+        if (!study.isExporter3Enabled()) {
+            exporter3Config = initExporter3ForStudy(appId, studyId);
+        } else {
+            exporter3Config = study.getExporter3Configuration();
+        }
 
         // Export the study's Timeline to Synapse as a wiki page in JSON format.
         JsonNode node = BridgeObjectMapper.get().valueToTree(timeline);
-
-//        File file = File.createTempFile("timelinefor" + studyId, ".txt");
-//        FileWriter fileWriter = new FileWriter(file);
-//        PrintWriter printWriter = new PrintWriter(fileWriter);
-//        printWriter.print(node.toString());
-//        printWriter.close();
-        PrintWriter pw = null;
-        File file = File.createTempFile("timelinefor" + studyId, ".txt");
+        File outputFile = File.createTempFile("timelineFor" + studyId, ".txt");
+        CharSink charSink = Files.asCharSink(outputFile, Charsets.UTF_8, FileWriteMode.APPEND);
         try {
-            FileOutputStream fos = new FileOutputStream(file);
-            pw = new PrintWriter(fos);
-            pw.print(node.toString());
-            pw.close();
-            pw = null;
-        } finally {
-            if (pw != null) {
-                pw.close();
-            }
+            charSink.write(node.toString());
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
If we have an IOException, the wiki page will almost certainly fail to upload. Instead of catching it, we should just let the exception get thrown.